### PR TITLE
Add road points reducing algorithm

### PR DESCRIPTION
**Pull request by aaravpandya**
This PR aims to do the following:

1. Reduce the number of sampled road points based on input threshold.
2. Export the road points vector for easy debugging.

This is not final PR. Some changes maybe made as per preferences -

1. I have added the speedup script with hardcoded path for dataset. This is because the previous configs style folder has been removed. The script can be moved and/or modified as per the project style.
2. Adding a flag for initializing graphics to check which road points to use for rendering is pending.

### CODE REVIEW

**Comment by daphnecor** [comment]
Hi Aarav,

Thanks for doing this, really excited to try this out!

I have two questions:
1. Have you tested whether it works correctly? Are there additional tests I can run on this end?
2. Is there a way to turn the road point reduction algorithm off completely? If there are bugs in it, we don't want it to influence my experiments. If so, can you add documentation/briefly explain it? 
3. It looks like this feature speeds up the simulator speed but does not reduce the observation space. Would it be possible to decrease the size of the observations through this as well? Now the observation probably contains more zeros.

**Comment by aaravpandya** [comment]
> 1. Have you tested whether it works correctly? Are there additional tests I can run on this end?
So I have tested it by running rendering. You can run it from here - https://github.com/facebookresearch/nocturne/blob/main/examples/rendering.py
However, I am going to take another look because I am not convinced that it is entirely correct.

> 2. Is there a way to turn the road point reduction algorithm off completely? If there are bugs in it, we don't want it to influence my experiments. If so, can you add documentation/briefly explain it?
Yes. You can set the reducing_threshold in the config to 0. Note the condition in the code to skip a point is using strictly less than  (<) operator. Since area can never be < 0, it will include all the points.

> 3. It looks like this feature speeds up the simulator speed but does not reduce the observation space. Would it be possible to decrease the size of the observations through this as well? Now the observations probably just contain more zeros. Tagging in @eugenevinitsky
I can take a look at how the observations are formed.

**Comment by eugenevinitsky** [comment]
Yeah, this won't change the observation size. We'd need to change the values of num_road_points in the observation config for that. We can pick a good value for that by looking at the renderings of the agent view once we pick a threshold

**Review by daphnecor — APPROVED** [review]

File: nocturne/cpp/include/scenario.h [hunk 1]
```diff
@@ -319,7 +321,7 @@ class Scenario : public sf::Drawable {
   // from the set of road points that comprise each polyline, we take
   // every n-th one
   const int64_t sample_every_n_ = 1;
-
+  const float_t reducing_threshold_ = 0.01;
```
Comment (eugenevinitsky) [inline nocturne/cpp/include/scenario.h:324]: hm, why is this value here? It seems like you are reducing always

**Pull request merged**